data/dataset.py

```diff
@@ -263,12 +267,20 @@ def fetch_n_hop_neighbourhood(
             new_edges = t.cat([x[1] for x in new_articles_and_edges], dim=1)
             accum_edges = t.cat([accum_edges, new_edges], dim=1)
 
-        articles_queue.extend(new_articles)
+        articles_queue.extend(
```

Comment:
why do we transform this to a tensor and transform it back? 
you can shuffle a list easily!

also, can we do this only when articles_queue's is larger than `num_neighbors`? otherwise we add a random shuffle operation unnecessarily (that we'll call a lot)